### PR TITLE
Change usage of window

### PR DESCRIPTION
**Pull request by elliothershberg**
This is a tiny PR takes some of the logic out of #1125 that changes how `window` is used. This is to help with running JB2 in other contexts than the browser, where `window` might not be defined.

This is related to #1786 . I'm looking into how our React LGV works inside of a vanilla Next.js app, and have run into issues with `window` not being defined.

### CODE REVIEW

File: packages/core/util/io/ElectronLocalFile.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (rbuels) [inline packages/core/util/io/ElectronLocalFile.ts:1]: not at the top of the file please ;-)

File: packages/core/util/io/ElectronRemoteFile.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ declare global {
     electron?: import('electron').AllElectron
   }
 }
-const { electron } = window
+const { electron } = typeof window !== 'undefined' ? window : ({} as any)
```
Comment (rbuels) [inline packages/core/util/io/ElectronRemoteFile.ts:18]: not at the top of the file please

**Comment by rbuels** [comment]
seems pretty straightforward, any reason this is still a draft?

**Comment by elliothershberg** [comment]
> seems pretty straightforward, any reason this is still a draft?

Was just checking to make sure it removed all lines causing the error I was seeing. Yeah, should be good to go.

File: packages/core/rpc/ElectronRpcDriver.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable no-await-in-loop */
+/* eslint-disable no-await-in-loop,@typescript-eslint/no-explicit-any */
```
Comment (rbuels) [inline packages/core/rpc/ElectronRpcDriver.ts:1]: not at the beginning of the file please

Comment (rbuels) [reply packages/core/rpc/ElectronRpcDriver.ts:1]: found one more

Comment (rbuels) [reply packages/core/rpc/ElectronRpcDriver.ts:1]: heh can you have an eslint rule to disable certain invocations of eslint-disable?

Comment (elliothershberg) [reply packages/core/rpc/ElectronRpcDriver.ts]: That is very meta 👽

**Pull request merged**